### PR TITLE
Send SMS with reminder email to passes which have phone numbers

### PR DESCRIPTION
**Pull request by Christopher-walsh22**
### Jira Ticket:

BRS-278

### Jira Ticket URL:

[BRS-278](https://github.com/orgs/bcgov/projects/49/views/2?pane=issue&itemId=42222375)

### Description:

- Added functionality to send SMS notifications to users through gcNotify
- Included SMS feature in the email notification sequence 
- Updated terraform file to look for new env-var in parameter store

### CODE REVIEW

File: lambda/smsUtil.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const AWS = require('aws-sdk');
+const { gcnSend } = require('./gcNotifyUtils');
+const { sendResponse } = require('./responseUtil');
+const { logger } = require('./logger');
+
+
+exports.sendSMSMessage = async function (payload, cancellationLink){
+    try {
+       let gcnSendObj = {
```
Comment (marklise) [inline lambda/smsUtil.js:9]: `const` not `let` since you are not modifying it later.

File: lambda/smsUtil.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        resData = res?.data?.response?.data;
+        jobError = 'SMS Notification failed: ';
+        logger.error(jobError, resData );
+        throw new Error('SMS Notification failed');
+      } else {
+        resData = res?.data?.data?.data;
+      }
+      logger.info(resData);
+  
+    return sendResponse(200, { msg: 'All works?', title: 'Completed actions' })
```
Comment (marklise) [inline lambda/smsUtil.js:31]: Indentation

**Review by marklise — APPROVED** [review]
👍 /w nits

File: lambda/smsUtil.js (hunk 1 above)
Comment (Christopher-walsh22) [reply lambda/smsUtil.js]: fixed in commit [0f1d4ee](https://github.com/bcgov/parks-reso-api/pull/330/commits/0f1d4eebe2a175a7de826b38d3a73f2b71648b2b)

File: lambda/smsUtil.js (hunk 2 above)
Comment (Christopher-walsh22) [reply lambda/smsUtil.js]: fixed in [0f1d4ee](https://github.com/bcgov/parks-reso-api/pull/330/commits/0f1d4eebe2a175a7de826b38d3a73f2b71648b2b)

**Review by cameronpettit — APPROVED** [review]
Approve w/nit - Before you merge, ensure all the AWS envs Parameter Stores have been updated with a value for ` data.aws_ssm_parameter.gc_notify_sms_template_id`. Looks like dev already has one but for the other environments, the deployment will bonk if those values aren't present. Better to do them now so you don't have to remember when we deploy to test/prod

**Comment by Christopher-walsh22** [comment]
I have updated Test and Prod with the required parameters! Thanks for the heads up.

**Pull request merged**